framework/PostProcessorFunctions/riskMeasuresDiscrete.py

```diff
@@ -109,58 +125,138 @@ def readMoreXML(self,xmlNode):
           self.target['low']  = min(val1,val2)
           self.target['high'] = max(val1,val2)
         else:
-          self.raiseAnError(IOError, 'RiskMeasuresDiscrete Interfaced Post-Processor ' + str(self.name) + ' : attribute node values is not present for XML node: ' + str(child) )
+          self.raiseAnError(IOError, 'RiskMeasuresDiscrete Interfaced Post-Processor ' + str(self.name) +
+                            ' : attribute node values is not present for XML node: ' + str(child) )
 
       elif child.tag == 'data':
         self.IEdata[child.text] = float(child.attrib['freq'])
 
+      elif child.tag == 'temporalID':
+        self.temporalID = child.text
+
       elif child.tag !='method':
-        self.raiseAnError(IOError, 'RiskMeasuresDiscrete Interfaced Post-Processor ' + str(self.name) + ' : XML node ' + str(child) + ' is not recognized')
+        self.raiseAnError(IOError, 'RiskMeasuresDiscrete Interfaced Post-Processor ' + str(self.name) +
+                          ' : XML node ' + str(child) + ' is not recognized')
 
-    if not set(self.measures).issubset(['B','FV','RAW','RRW']):
+    if not set(self.measures).issubset(self.availableMeasures()):
+      unrecognizedMeasures  = set(self.measures).difference(self.availableMeasures())
       self.raiseAnError(IOError, 'RiskMeasuresDiscrete Interfaced Post-Processor ' + str(self.name) + ' : measures '
-                        + str(set(self.measures).issubset([B,FV,RAW,RRW])) + ' are not recognized')
+                        + str(list(unrecognizedMeasures)) + ' are not recognized')
+
 
   def run(self,inputDic):
     """
      This method perform the actual calculation of the risk measures
      @ In, inputDic, list, list of dictionaries which contains the data inside the input DataObjects
      @ Out, outputDic, dict, dictionary which contains the risk measures
     """
+    # Check how many HistorySets (checkHSs) have been provided
+    checkHSs=0
+    for inp in inputDic:
+      if inp['type'] == 'HistorySet':
+        timeDepData = copy.deepcopy(inp)
+        inputDic.remove(inp)
+        checkHSs +=1
+
+    if checkHSs == 0:
+      # if no HistorySet has been provided run the static form of this PP
+      outputDic = self.runStatic(inputDic)
+    elif checkHSs == 1:
+       # if one HistorySet has been provided run the dynamic form of this PP
+      self.outputFormat = 'HistorySet'
+      outputDic = self.runDynamic(inputDic,timeDepData)
+      for var in timeDepData['data']['output'][1]:
+        if var != self.temporalID:
+          if not np.array_equal( np.union1d( np.unique(timeDepData['data']['output'][1][var]) , np.array([0,1])) , np.array([0,1]) ):
```

Comment:
What is this line doing? Are you trying to test if the only values in timeDepData['data']['output'][1][var] are zero and/or one? 

This is a clever line, but is there any more straightforward (and efficient) way to do this? How about setdiff1d?

```python
## Are there any values in timeDepData['data']['output'][1][var] that are not 0 or 1?
if len(np.setdiff1d(timeDepData['data']['output'][1][var], [0,1])):
```